lib/Sema/CSDiag.cpp

```diff
@@ -654,6 +654,67 @@ static bool isLoadedLValue(Expr *expr) {
   return false;
 }
 
+static bool appendPatternTypeWithVarReplaced(
+                                             SmallString<8> &scratch, Pattern *P,
+                                             VarDecl *VD, Type vdType) {
+  struct AppendReplacedType : PatternVisitor<AppendReplacedType, bool> {
+    VarDecl *VD;
+    Type VDType;
+    SmallString<8> &scratch;
+    
+    AppendReplacedType(VarDecl *VD, Type vdType, 
+                       SmallString<8> &scratch)
+    : ASTVisitor(), VD(VD), VDType(vdType), scratch(scratch)
+    {}
+    
+    bool visitTuplePattern(TuplePattern *TP) {
+      scratch += '(';
+      for (auto i : indices(TP->getElements())) {
+        if (i != 0) scratch.append(", ");
+        if (!visit(TP->getElement(i).getPattern()))
+          return false;
+      }
+      scratch += ')';
+      return true;
+    }
+    bool visitParenPattern(ParenPattern *PP) {
+      scratch += '(';
+      if (!visit(PP->getSubPattern()))
+        return false;
+      scratch += ')';
+      return true;
+    }
+    bool visitOptionalSomePattern(OptionalSomePattern *OSP) {
+      if (!visit(OSP->getSubPattern()))
+        return false;
+      scratch += '?';
+      return true;
+    }
+    
+    bool visitVarPattern(VarPattern *P) { return visit(P->getSubPattern()); }
```

Comment:
I think you can mark all these `case`-only patterns as `llvm_unreachable` instead.